src/engine/test/source/hlp/hlp_test.cpp

```diff
@@ -417,3 +417,220 @@ TEST(hlpTimestampTests, specific_format)
+    result = parseOp(relative_path);
+    ASSERT_EQ("Desktop\\test.txt", result["_file.path"]);
+    ASSERT_EQ("", result["_file.drive_letter"]);
+    ASSERT_EQ("Desktop", result["_file.folder"]);
+    ASSERT_EQ("test.txt", result["_file.name"]);
+    ASSERT_EQ("txt", result["_file.extension"]);
+
+    const char *file_without_ext = "Desktop\\test";
+    result = parseOp(file_without_ext);
+    ASSERT_EQ("Desktop\\test", result["_file.path"]);
+    ASSERT_EQ("", result["_file.drive_letter"]);
+    ASSERT_EQ("Desktop", result["_file.folder"]);
+    ASSERT_EQ("test", result["_file.name"]);
+    ASSERT_EQ("", result["_file.extension"]);
+
+    const char *only_file = "test.txt";
+    result = parseOp(only_file);
+    ASSERT_EQ("test.txt", result["_file.path"]);
+    ASSERT_EQ("", result["_file.drive_letter"]);
+    ASSERT_EQ("", result["_file.folder"]);
+    ASSERT_EQ("test.txt", result["_file.name"]);
+    ASSERT_EQ("txt", result["_file.extension"]);
+
+    const char *folder_path = "C:\\Users\\Name\\Desktop\\";
+    result = parseOp(folder_path);
+    ASSERT_EQ("C:\\Users\\Name\\Desktop\\", result["_file.path"]);
+    ASSERT_EQ("C", result["_file.drive_letter"]);
+    ASSERT_EQ("C:\\Users\\Name\\Desktop", result["_file.folder"]);
+    ASSERT_EQ("", result["_file.name"]);
+    ASSERT_EQ("", result["_file.extension"]);
+
+    const char *lower_case_drive = "c:\\test.txt";
+    result = parseOp(lower_case_drive);
+    ASSERT_EQ("c:\\test.txt", result["_file.path"]);
+    ASSERT_EQ("C", result["_file.drive_letter"]);
+    ASSERT_EQ("c:", result["_file.folder"]);
+    ASSERT_EQ("test.txt", result["_file.name"]);
+    ASSERT_EQ("txt", result["_file.extension"]);
+}
+
+TEST(filepath_test, unix_path)
+{
+    const char *logQl ="<_file/FilePath>";
+    ParserFn parseOp = getParserOp(logQl);
+    ParseResult result;
+
+    const char *full_path = "/Desktop/test.txt";
+    result = parseOp(full_path);
+    ASSERT_EQ("/Desktop/test.txt", result["_file.path"]);
+    ASSERT_EQ("", result["_file.drive_letter"]);
+    ASSERT_EQ("/Desktop", result["_file.folder"]);
+    ASSERT_EQ("test.txt", result["_file.name"]);
+    ASSERT_EQ("txt", result["_file.extension"]);
+
+    const char *relative_path = "Desktop/test.txt";
+    result = parseOp(relative_path);
+    ASSERT_EQ("Desktop/test.txt", result["_file.path"]);
+    ASSERT_EQ("", result["_file.drive_letter"]);
+    ASSERT_EQ("Desktop", result["_file.folder"]);
+    ASSERT_EQ("test.txt", result["_file.name"]);
+    ASSERT_EQ("txt", result["_file.extension"]);
+
+    const char *folder_path = "/Desktop/";
+    result = parseOp(folder_path);
+    ASSERT_EQ("/Desktop/", result["_file.path"]);
+    ASSERT_EQ("", result["_file.drive_letter"]);
+    ASSERT_EQ("/Desktop", result["_file.folder"]);
+    ASSERT_EQ("", result["_file.name"]);
+    ASSERT_EQ("", result["_file.extension"]);
+}
+
+TEST(filepath_test, force_unix_format)
+{
+    const char *logQl ="<_file/FilePath/UNIX>";
```

Comment:
It could be a real scenario. But, at the moment, there isn´t a specification of an ECS field that must be completely parsed as a FilePath. Thus it isn´t included in the kECSParserMapper.
Anyway, this structure will be replaced once a full asset is read to determine the fields/parser association.